Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-contest-categories-breadcrumb.tsx

```diff
@@ -0,0 +1,105 @@
+﻿import React, { createContext, useCallback, useContext, useState } from 'react';
+import { isNil } from 'lodash';
+import { IContestCategoryTreeType } from '../../common/types';
+import { ITreeItemType } from '../../components/guidelines/trees/Tree';
+import { IHaveChildrenProps } from '../../components/common/Props';
+
+interface ICategoriesBreadcrumbContext {
+    state: {
+        breadcrumbItems: ICategoriesBreadcrumbItem[];
+    };
+    actions: {
+        updateBreadcrumb: (category: ITreeItemType | undefined, flattenTree: ITreeItemType[] | []) => void;
+    };
+}
+
+interface ICategoriesBreacrumbProviderProps extends IHaveChildrenProps {
+}
+
+interface ICategoriesBreadcrumbItem {
+    id: string,
+    isLast: boolean,
+    value: string,
+    orderBy: number,
+}
+
+const defaultState = { state: { breadcrumbItems: [] as ICategoriesBreadcrumbItem[] } };
+
+const CategoriesBreadcrumbContext = createContext<ICategoriesBreadcrumbContext>(defaultState as ICategoriesBreadcrumbContext);
+
+const orderByAsc = (x : ICategoriesBreadcrumbItem, y: ICategoriesBreadcrumbItem) => y.orderBy - x.orderBy;
+
+const CategoriesBreadcrumbProvider = ({ children }: ICategoriesBreacrumbProviderProps) => {
```

Comment:
Does this need so descriptive name? 
Also, isn't this logic for `useContestCategories`?

Reply:
Not sure at all. I had implemented it in useContestCategories but then I thought isn't it better to separate it because of the separation of responsibilities between the hooks and to be more readable and easy to understand. It can be moved into useContestCategories also it makes sense. If you think it would be better?